andela/Ah-backend-xmen
# #162725272 Users should be able to bookmark articles

#### What does this PR do?

This PR introduces functionality for users to bookmark articles. Every logged in user has a read later list where bookmarks are kept.

#### Description of Task to be completed?
create 2 new endpoints
`GET /api/bookmarks`
returns a list of bookmarked articles
`PUT /api/articles/<slug>/bookmark`
Toggle add to bookmarks and remove from bookmarks

#### How should this be manually tested?
- fetch this branch by running the command `git fetch origin ft-bookmark-read-later-162725272` in the terminal.
- check into the branch `git checkout ft-bookmark-read-later-162725272`
- start the virtual environment `source venv/bin/activate`
- migrate to update the database using `python manage.py migrate`
-  run development server using `python manage.py runserver`
#### What are the relevant pivotal tracker stories?
[#162725272](https://www.pivotaltracker.com/story/show/162725272)

#### Screenshots (if appropriate)
<img width="951" alt="screen shot 2019-01-23 at 12 07 20 pm" src="https://user-images.githubusercontent.com/17139402/51595345-8e0d9e00-1f07-11e9-9a5b-5d7d7e1610f2.png">

<img width="916" alt="screen shot 2019-01-23 at 12 08 31 pm" src="https://user-images.githubusercontent.com/17139402/51595373-a4b3f500-1f07-11e9-8a5b-9bf3aa4c2bf3.png">

<img width="972" alt="screen shot 2019-01-23 at 12 09 01 pm" src="https://user-images.githubusercontent.com/17139402/51595415-b4cbd480-1f07-11e9-8dc9-353650d98384.png">

#### Postman Documentation

[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/dce385f165d7dcf93c6b)
